blog/en/blog/2025/03/31/march-monthly-report.md

```diff
@@ -0,0 +1,120 @@
+
+![Apache APISIX Contributors List](https://static.api7.ai/uploads/2025/03/31/fDYTi9Vu_march-contributor-list.webp)
+
+![New Contributors List](https://static.api7.ai/uploads/2025/03/31/NB9qpu7Q_mar-new-contributors.webp)
+
+## Good First Issue
+
+### Issue #12098
+
+Link：[https://github.com/apache/apisix/issues/12098](https://github.com/apache/apisix/issues/12098)
+
+Problem Description: In [PR #12029](https://github.com/apache/apisix/pull/12029), the configuration code for the plugin needs to be improved.
+
+## Feature Highlights
+
+### New Plugins
+
+**1. [Support `ai-prompt-guard` Plugin](https://github.com/apache/apisix/pull/12008)(Contributor: [Revolyssup](https://github.com/Revolyssup))**
+
+The `ai-prompt-guard` plugin safeguards your AI endpoints by inspecting and validating incoming prompt messages. It checks the content of requests against user-defined allowed and denied patterns to ensure that only approved inputs are processed. Based on its configuration, the plugin can either examine just the latest message or the entire conversation history, and it can be set to check prompts from all roles or only from end users.
+
+**2. [Support `ai-rate-limiting` Plugin](https://github.com/apache/apisix/pull/12037)(Contributor: [shreemaan-abhishek](https://github.com/shreemaan-abhishek))**
+
+The `ai-rate-limiting` plugin enforces token-based rate limiting for requests sent to LLM services. It helps manage API usage by controlling the number of tokens consumed within a specified time frame, ensuring fair resource allocation and preventing excessive load on the service. It is often used with `ai-proxy-multi` plugin.
+
+**3. [Support `ai-request-rewrite` Plugin](https://github.com/apache/apisix/pull/12036)(Contributor: [LiteSun](https://github.com/LiteSun))**
+
+The ai-request-rewrite plugin leverages predefined prompts and AI services to intelligently modify client requests, enabling AI-powered content transformation before forwarding to upstream services.
+
+### New/Enhanced Features
+
+**1. [Support Proxying OpenAI-Compatible LLMs](https://github.com/apache/apisix/pull/12004)(Contributor: [shreemaan-abhishek](https://github.com/shreemaan-abhishek))**
+
+This PR introduces a new provider, `openai-compatible`, that can proxy requests to OpenAI-compatible LLMs.
+
+**2. [Store JWT in the Request Context in `jwt-auth` Plugin](https://github.com/apache/apisix/pull/11675)(Contributor: [mikyll](https://github.com/mikyll))**
+
+The `jwt-auth` plugin now has a new parameter: `store_in_ctx`. When enabled (default is off), it stores the validated JWT object in the request context.
+
+This feature is especially beneficial for custom plugins:
+
+- When `hide_credential = true` removes the JWT from request attributes, store_in_ctx offers a secure alternative for token passing without exposure.
+
+- It prevents custom plugins from duplicating code to retrieve and parse JWT objects, as the `jwt-auth` plugin already handles these tasks.
```

Comment:
> It prevents custom plugins from duplicating code to retrieve 

can't understand, please rephrase

Reply:
It prevents custom plugins from duplicated code, for example, retrieve and parse JWT objects.

Reply:
"prevents custom plugins from duplicated code" makes no sense ah

Reply:
delete this reason?